components/TokenApprovalList/index.tsx

```diff
@@ -267,20 +197,31 @@ const TokenApprovalList: React.FC<TokenApprovalListProps & { maxCount?: string;
     )
   }
 
-  const isOwner = connectedAddr ? (id as string).toLowerCase() === connectedAddr.toLowerCase() : true
-  const disabled = !isOwner || isPreparingContract || !connector.ready || isRevokeTxnLoading
+  // hide and cache:
+  // - the listitem has revoke txn packaged successfully in a block but backend not synced yet
+  const hideItem = (txnHash: string) => {
```

Comment:
Index the record by `tx hash` may be not enough because a transaction could have multiple approvals, e.g. https://v1.gwscan.com/tx/0x26b3c6f90a9756cbc9e10a820d3519cdf86523790e1eb77b0e84aca558008bdc?tab=logs

Reply:
changed to use `transaction_hash + spender_address_hash` as cache key, is this enough to distinguish entries?

Reply:
I think `contract address` and `token id`(if exist) should also be indexed because I can approve a spender to manage many of my nfts in one transaction.

For erc20, I can also approve a spender to manage multiple erc20 tokens in one transaction